src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQueryDatasetPdao.java

```diff
@@ -1428,4 +1430,35 @@ private static String bqStringValue(FieldValueList fieldValue, String fieldName)
     }
     return null;
   }
+
+  /// Snapshot Builder
+  // Pass in SQL query and map into generic BigQuery TableResult
+  public <T> List<T> runSnapshotBuilderQuery(
+      String sql, Dataset dataset, Function<TableResult, List<T>> aggregateSnapshotBuilderResults) {
+    try {
+      final BigQueryProject bigQueryProject = BigQueryProject.from(dataset);
+      final TableResult result = bigQueryProject.query(sql);
+      return aggregateSnapshotBuilderResults.apply(result);
+    } catch (InterruptedException ex) {
+      throw new PdaoException("Snapshot builder query was interrupted", ex);
+    }
+  }
+
+  // Convert BigQuery TableResult into List<SnapshotBuilderConcept>
+  public Function<TableResult, List<SnapshotBuilderConcept>>
```

Comment:
Can this be defined in the snapshot builder code? Unless we think non-snapshot builder code will need to run this API. It seems to me that `BigQueryDatasetPdao` shouldn't need to depend on `SnapshotBuilderConcept`

Reply:
Yes, it could. But, I think this code is rather specific to the cloud specific pdao. 
With my last commit, I propose an alternative of putting this in a util within the snapshotbuilder folder. We expect for there to be multiple aggregation methods, so we can put them all in the util, but still group by cloud platform.